pandas/tests/test_resample.py

```diff
@@ -2799,89 +2693,182 @@ def test_evenly_divisible_with_no_extra_bins(self):
         result = df.resample('7D').sum()
         assert_frame_equal(result, expected)
 
-    def test_loffset_returns_datetimeindex(self):
+
+def assert_series_or_frame_equal(result, expected):
+    if isinstance(result, Series):
+        return assert_series_equal(result, expected)
+    else:
+        return assert_frame_equal(result, expected)
+
+
+class TestPeriodIndexIdiomaticPytest:
+
+    @pytest.fixture(scope='class')
+    def index(self):
+        return period_range(datetime(2005, 1, 1),
+                            datetime(2005, 1, 10), freq='D')
+
+    @pytest.fixture(scope='class')
+    def series(self, index):
+        return Series(np.arange(len(index)), index=index, name='pi')
+
+    @pytest.fixture(scope='class')
+    def frame(self, index):
+        return DataFrame({'value': np.arange(len(index))}, index=index)
+
+    @pytest.fixture(params=[Series, DataFrame], scope='class')
+    def pandas_obj(self, request, index):
```

Comment:
tried another approach by keeping the existing inheritance structure - we can inherit fixtures defined in the ```Base``` class, ```index``` fixture defined in `Base` dispatches to the subclass-specific type via the existing ```_index_factory```. If you want to take a look:
https://github.com/winklerand/pandas/blob/bbf01b312a2d8be0b3c7681b9a46a039f33fdcc/pandas/tests/test_resample_pytest.py

Seems easier to me than parametrizing the index fixture, there's a lot of index type-specific test logic.

Should we move out this discussion to another issue or PR? This looks like a more general issue on how to (gradually) transition to pytest.